lib/templateFactory.js

```diff
@@ -270,9 +271,13 @@ class TemplateFactory extends BaseFactory {
                 aggregationField: 'templateId'
             };
 
-            const jobListConfig = {
-                params: { templateId: templateIds },
-                readOnly: true
+            const jobsQueryConfig = {
+                queries: getQueries(this.datastore.prefix, PIPELINE_USAGE_COUNT),
+                readOnly: true,
+                replacements: {
+                    templateIds
+                },
+                rawResponse: true
             };
```

Comment:
Abstract this into a new method in `JobFactory`

```suggestion
// this code goes into /lib/jobFactory.js
getPipelineUsageCountForTemplates(templateIds) {
    const jobsQueryConfig = {
                queries: getQueries(this.datastore.prefix, PIPELINE_USAGE_COUNT),
                readOnly: true,
                replacements: {
                    templateIds
                },
                rawResponse: true
            };
            
    return jobFactory.query(jobsQueryConfig).then(r => return r[0]);
}
            
```